### DISCUSSION TITLE
Any way to automatically merge folders?

### DISCUSSION THREAD

**Discussion by loiteringsloth** [WinMerge/winmerge#1130] [Q&A]
Is manually selecting each file/folder the only way to merge? 

I a have a folder and an external backup. Folder size is over 900gb, acutal difference between the two is <30gb, but most subfolders are quite large, larger than available free space.

Windows wants the entire folder size to be available as if I'm copying the whole thing instead of merging.

I've done the folder comparison using WinMerge, but as far as I can see it can only highlight differences and copy one single file or folder at a time.

Is there any way to tell WinMerge to automatically copy all the files missing to the backup folder only?

Thanks

*tried switching to expanded view, selecting all from left only, then copy to right. Windows file copy window thing opens, discovers over 800GB to be copied (not what I selected) then it crashes and I get an 'operation failed' message

**Comment by sdottaka** [comment]
> Is there any way to tell WinMerge to automatically copy all the files missing to the backup folder only?

> Is there any way to tell WinMerge to automatically copy all the files missing to the backup folder only?

Unfortunately, WinMerge does not have the ability to automatically merge files/folders.
You may want to try other tools such as FreeFileSync or RoboCopy.

Here's how to copy unique files manually using WinMerge: 

1. Uncheck the View -> TreeMode menu item. (Display in flat mode is very slow. 2.16.17 beta improves it a bit.)
2. Sort by Comparison result column. 
3. Select the Left only or Right only items. 
4. Right Click → Copy → Left to Right or Right to Left

> *tried switching to expanded view, selecting all from left only, then copy to right. Windows file copy window thing opens, discovers over 800GB to be copied (not what I selected) copied (not what I selected) then it crashes and I get an 'operation failed' message

Please let us know if you know the steps or conditions we can reproduce the problem.

**Reply by loiteringsloth** [reply]
Ah, I see 

Thanks for the recommendations, looks freefilesync does exactly what I need!
